tests/conftest.py

```diff
@@ -0,0 +1,37 @@
+import json
+import warnings
+
+from elasticsearch import Elasticsearch, exceptions
+from elasticsearch.client import IndicesClient
+import pytest
+
+import ingest.bulk_upload
+
+
+@pytest.fixture(scope="session")
+def test_index_name():
+    """The name of the fake index to use throughout the tests."""
+    return "oracc_test"
+
+
+@pytest.fixture
+def es(monkeypatch, test_index_name):
+    """An ElasticSearch client which also ensures we operate on a fake index."""
+    # NB: INDEX_NAME cannot be imported directly, or the patching won't work.
+    # Also note that this fixture cannot be module-scoped because monkeypatch is
+    # function-scoped.
+    monkeypatch.setattr(ingest.bulk_upload, "INDEX_NAME", test_index_name)
+    assert ingest.bulk_upload.INDEX_NAME == test_index_name  # just making sure
```

Comment:
It has taken me a little while and some StackOverflow research to feel OK about an assert being inside of a fixture.

Reply:
I've just tested it myself: a bad assertion in a fixture makes all tests that use it fail, before any code in those tests is run. I think it's safe enough to leave in!